File: feign/src/main/java/br/com/leonardoferreira/poc/feign/client/TodoClient.java
package br.com.leonardoferreira.poc.feign.client;

import br.com.leonardoferreira.poc.feign.domain.Todo;
import java.util.List;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(
        name = "todoClient",
        url = "${jsonplaceholder.url}/todos"
)
public interface TodoClient {

    @GetMapping("/{id}")
    Todo findById(@PathVariable("id") Long id);

    @GetMapping
    List<Todo> findByUserId(@RequestParam("userId") Long userId);

}

File: feign/parallel/src/main/kotlin/br/com/leonardoferreira/poc/client/TodoClient.kt
package br.com.leonardoferreira.poc.client

import br.com.leonardoferreira.poc.domain.response.TodoResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable

@FeignClient(
    name = "todoClient",
    url = "https://jsonplaceholder.typicode.com/"
)
interface TodoClient {

    @GetMapping("/todos/{id}")
    fun findById(@PathVariable id: Long): TodoResponse

}
